flexmeasures/data/models/reporting/cost.py

```diff
@@ -0,0 +1,143 @@
+
+    Sign convention
+    ----------------
+
+    Power flows:
+        (+) production
+        (-) consumption
+
+    Cash flows:
+        (+) to charge
+        (-) to pay
+
+    """
+
+    __version__ = "1"
+    __author__ = "Seita"
+
+    _config_schema = CostReporterConfigSchema()
+    _parameters_schema = CostReporterParametersSchema()
+
+    weights: dict
+    method: str
+
+    def _compute_report(
+        self,
+        start: datetime,
+        end: datetime,
+        input: List[Dict[str, Any]],
+        output: List[Dict[str, Any]],
+        resolution: timedelta | None = None,
+        belief_time: datetime | None = None,
+    ) -> List[Dict[str, Any]]:
+        """
+        :param start: start time of the report
+        :param end: end time of the report
+        :param input: power/energy sensor to consider
+        :param output: sensor where to save the report to. Specify multiple
+                       output sensors with different resolutions to save
+                       the results in multiple time frames (e.g. hourly, daily).
+        :param resolution: _description_, defaults to None
+        :param belief_time: time where the information is available.
+        """
+
+        production_price_sensor: Sensor = self._config.get("production_price_sensor")
+        consumption_price_sensor: Sensor = self._config.get("consumption_price_sensor")
+
+        input_sensor: Sensor = input[0]["sensor"]  # power or energy sensor
+        timezone = input_sensor.timezone
+
+        if belief_time is None:
+            belief_time = server_now()
+
+        # get prices
+        production_price = simplify_index(
+            production_price_sensor.search_beliefs(
+                event_starts_after=start,
+                event_ends_before=end,
+                beliefs_before=belief_time,
+                resolution=input_sensor.event_resolution,
+            )
+        )
+        production_price = production_price.tz_convert(timezone)
+        consumption_price = simplify_index(
+            consumption_price_sensor.search_beliefs(
+                event_starts_after=start,
+                event_ends_before=end,
+                beliefs_before=belief_time,
+                resolution=input_sensor.event_resolution,
+            )
+        )
+        consumption_price = consumption_price.tz_convert(timezone)
+
+        # get power/energy time series
+        power_energy_data = simplify_index(
+            input_sensor.search_beliefs(
+                event_starts_after=start,
+                event_ends_before=end,
+                resolution=resolution,
+                beliefs_before=belief_time,
+            )
+        )
+
+        # compute energy flow from power flow
+        if input_sensor.measures_power:
+            power_energy_data *= input_sensor.event_resolution / timedelta(hours=1)
+
+        # compute cashflow
+        # this step assumes that positive flows represent production and negative flows consumption
```

Comment:
We should check the meaning of the sign of values stored under the power/energy sensor, using the `consumption_is_positive` attribute.

Reply:
Good one!